netlogo-web/src/test/scala/TortoiseFinder.scala

```diff
@@ -204,6 +205,17 @@ private[tortoise] object Freebies {
     "Run::run-evaluate-string-input-only-once"
   )
 
+  private val importWorldStr = "Tortoise does not know how to import from a filepath (see: JavaScript)"
+  private val importWorldCommandNames = Seq(
+    // Uncomment these if `__mkdir` is ever implemented.  --JAB (2/7/18)
+    // "ImportWorld::AllTurtlesExport",
+    // "ImportWorld::AllBreedsExport",
+    // "ImportWorld::AllLinkBreedsExport",
+    // "ImportWorld::ImportExportBreedOrder",
+    // "ImportWorld::ImportExportLinkBreedOrder",
```

Comment:
Could these tests be written in a way that doesn't require `__mkdir`? I'm open to changing desktop tests if needed. Alternatively, could we implement a `__tmpdir` primitive which creates and gives the name of a temporary "directory" (obviously, browser javascript can't create directories, but it might be good enough to return the downloads directory or something, idk)?

Reply:
Thanks for this comment.  As I was writing to explain to you the reasoning behind all of this, I came to realize that these lines of code weren't necessary at all.  Consequently, I've updated the PR (diffs between the old and new versions of the PR (exempting dumps) can be found [here](https://gist.github.com/TheBizzle/314fc8ddc87ff8db93cc474e593fce4f)).

What it essentially boiled down to was that I had it in my head that the language tests wouldn't be able to do `import-world` language tests since JS won't let us read files synchronously, but, in Nashorn, we *can* read the files synchronously, if only we take that `java.nio`/`fileReader` stuff that you noticed and actually *use* it (rather than declaring it and then just ignoring it).

As for your actual question, absent a virtual filesystem or new `__` primitives for reading and writing `export-world` text without file interactions, I'm not seeing a way that those tests could really come to use a `__tmpdir` prim or otherwise avoid `__mkdir`.  It seems to me that that's just the nature of "testing `import-world` and `export-world`. 

Of course, after the changes that I made to this PR, that's all been rendered moot for `import-world`/`export-world`, since I've removed the test "freebies" that were commented on here.

Reply:
🎉